ZelBack/src/services/appsService.js

```diff
@@ -10055,28 +10064,42 @@ async function checkMyAppsAvailability() {
     const min = minPort;
     const max = maxPort;
     testingPort = failedPort || Math.floor(Math.random() * (max - min) + min);
-    log.info(`checkMyAppsAvailability testing port ${testingPort}.`);
-    const iBP = fluxNetworkHelper.isPortBanned(testingPort);
+
+    log.info(`checkMyAppsAvailability - Testing port ${testingPort}.`);
+    let iBP = fluxNetworkHelper.isPortBanned(testingPort);
     if (iBP) {
+      log.info(`checkMyAppsAvailability - Testing port ${testingPort} is banned.`);
       failedPort = null;
       // skip this check, port is not possible to run on flux
       await serviceHelper.delay(15 * 1000);
       checkMyAppsAvailability();
       return;
     }
+    if ((userconfig.initial.apiport && userconfig.initial.apiport !== config.server.apiport) || isUPNP) {
+      iBP = fluxNetworkHelper.isPortUPNPBanned(testingPort);
+      if (iBP) {
+        log.info(`checkMyAppsAvailability - Testing port ${testingPort} is UPNP banned.`);
+        failedPort = null;
+        // skip this check, port is not possible to run on flux
+        await serviceHelper.delay(15 * 1000);
+        checkMyAppsAvailability();
+        return;
+      }
+    }
     const isPortUserBlocked = fluxNetworkHelper.isPortUserBlocked(testingPort);
     if (isPortUserBlocked) {
+      log.info(`checkMyAppsAvailability - Testing port ${testingPort} is user blocked.`);
       failedPort = null;
       // skip this check, port is not allowed for this flux node by user
       await serviceHelper.delay(15 * 1000);
       checkMyAppsAvailability();
       return;
     }
     if (appPorts.includes(testingPort)) {
-      log.warn(`Skipped checking ${testingPort} - in use.`);
+      log.info(`checkMyAppsAvailability - Skipped checking ${testingPort} - in use.`);
       failedPort = null;
       // skip this check
-      await serviceHelper.delay(2 * 60 * 1000);
+      await serviceHelper.delay(15 * 1000);
```

Comment:
intentional?

Reply:
yes, the other if's clauses had a 15 seconds await for the same kind of behaviour, the random por selected can't be used for testing. So await 15 seconds and call the method again is good, 2 minutes is a lot.